src/backend/gl/src/window/egl.rs

```diff
@@ -364,32 +410,59 @@ impl hal::Instance<crate::Backend> for Instance {
         attributes.push(egl::ATTRIB_NONE);
 
         let native_window_ptr = match has_handle.raw_window_handle() {
+            #[cfg(target_os = "android")]
+            Rwh::Android(_) => native_window as *mut _ as _,
```

Comment:
why do we even need `native_window` produced in the first match, given that it's only used here in that second match over the same thing? It looks to me that there should only be one match, but we can follow-up separately...